src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -81,7 +79,7 @@ static FileSystemWatcher()
         public FileSystemWatcher()
         {
             _directory = string.Empty;
-            _filter = "*";
+            Filters.Add("*");
```

Comment:
Don't do this. You shouldn't get `{ "*", "foo" }`  when adding a new filter. Add tests.

Reply:
wont be we facing the same thing for ```public FileSystemWatcher(string path) : this(path, "*")``` should we change this also ?